R/run_stress_test.R

```diff
@@ -168,38 +168,18 @@ run_stress_test_impl <- function(args_list) {
     lgd_subordinated_claims = lgd_subordinated_claims
   )
 
-  scenarios_filter <- unique(
-    c(
-      baseline_scenario_lookup,
-      shock_scenario_lookup
-    )
-  )
+  cat("-- Reading input data from designated input path. \n")
+  data <- st_read(input_path_project_specific, asset_type)
 
-  cat("-- Importing and preparing input data from designated input path. \n")
-
-  pacta_results <- read_pacta_results(
-    path = file.path(input_path_project_specific, paste0(stringr::str_to_title(asset_type), "_results_", calculation_level_lookup, ".rda"))
-  )
-
-  start_year <- min(pacta_results$year, na.rm = TRUE)
-
-  pacta_results <- pacta_results %>%
-    process_pacta_results(
-      start_year = start_year,
-      end_year = end_year_lookup,
-      time_horizon = time_horizon_lookup,
-      scenario_geography_filter = scenario_geography_filter_lookup,
-      scenarios_filter = scenarios_filter,
-      equity_market_filter = equity_market_filter_lookup,
-      sectors = sectors_lookup,
-      technologies = technologies_lookup,
-      allocation_method = allocation_method_lookup,
-      asset_type = asset_type
-    )
+  cat("-- Processing input data. \n")
+  pacta_results <- st_process(data, asset_type)$pacta_results
 
   sector_exposures <- read_sector_exposures(file.path(input_path_project_specific, "overview_portfolio.rda")) %>%
     process_sector_exposures(asset_type = asset_type)
 
+  start_year <- get_start_year(data)
+  scenarios_filter <- scenarios_filter()
```

Comment:
These variables will go away once we extract each `read_*()` and `process_*()`. For now they keep the diff minimal.